src/program.js

```diff
@@ -63,14 +62,19 @@ export class Program {
     this.yargs.strict();
 
     this.commands = {};
-    this.defaultValues = {};
+    this.options = {};
+    this.options.mainCommandsList = [];
   }
 
   command(
     name: string, description: string, executor: Function,
     commandOptions: Object = {}
   ): Program {
-    this.defaultValues[name] = setDefaultValues(commandOptions);
+    const mainCommandsList = [];
+    mainCommandsList.push(name);
+    this.options = Object.assign(this.options, commandOptions);
+    this.options.mainCommandsList = mainCommandsList;
```

Comment:
I still think you can get it working with just this:

```js
this.options[name] = commandOptions;
```
All the data is there. Try to see if you can get it working.